packages/core/src/components/graphs/scatter.ts

```diff
@@ -106,13 +106,22 @@ export class Scatter extends Component {
 	addLabelsToDataPoints(d, index) {
 		const { labels } = this.model.getDisplayData();
 
-		return d.data.map((datum, i) => ({
-			date: datum.date,
-			label: labels[i],
-			datasetLabel: d.label,
-			class: datum.class,
-			value: isNaN(datum) ? datum.value : datum
-		}));
+		return d.data
+			// Remove datapoints with no value
+			.filter((datum: any) => {
+				if (!datum || datum.value === null) {
+					return false;
+				}
+
+				return true;
+			})
+			.map((datum, i) => ({
+				date: datum.date,
+				label: labels[i],
+				datasetLabel: d.label,
+        class: datum.class,
+				value: isNaN(datum) ? datum.value : datum
```

Comment:
I think `datum` is always an object here, is that right?

Reply:
Not always, if you're using a discrete scale then datum will just be a number.

Reply:
That's what I initially thought, but then I realized there is the access to `datum.class` in the line above. Should that be checked then?